dissect/target/plugins/apps/remoteaccess/splashtop.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import json
+import re
+from typing import TYPE_CHECKING
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.helpers.utils import year_rollover_helper
+from dissect.target.plugin import export
+from dissect.target.plugins.apps.remoteaccess.remoteaccess import (
+    GENERIC_FILE_TRANSFER_RECORD_FIELDS,
+    GENERIC_LOG_RECORD_FIELDS,
+    RemoteAccessPlugin,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    from dissect.target.helpers.fsutil import TargetPath
+    from dissect.target.target import Target
+
+
+RE_TS = re.compile(r"(\w{3}\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3})")
+RE_LOG_LINE = re.compile(r"<[0-9]>\w{3}\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3}\s(.*)")
+
+
+class SplashtopPlugin(RemoteAccessPlugin):
+    """Splashtop plugin."""
+
+    __namespace__ = "splashtop"
+
+    RemoteAccessLogRecord = TargetRecordDescriptor(
+        "remoteaccess/splashtop/log",
+        GENERIC_LOG_RECORD_FIELDS,
+    )
+
+    RemoteAccessFileTransferRecord = TargetRecordDescriptor(
+        "remoteaccess/splashtop/filetransfer",
+        GENERIC_FILE_TRANSFER_RECORD_FIELDS,
+    )
+
+    LOG_PATHS = (
+        "sysvol/Program Files (x86)/Splashtop/Splashtop Remote/Server/log/SPLog.txt",  # General agent log including connections and filetransfers
+        # "sysvol/ProgramData/Splashtop/Temp/log/FTCLog.txt", # File transfer log which is included in the SPLog
+    )
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+
+        self.log_files: set[TargetPath] = set()
+
+        for log_path in self.LOG_PATHS:
+            log_file = self.target.fs.path(log_path)
+            if log_file:
+                self.log_files.add(log_file)
+
+    def check_compatible(self) -> None:
+        if not self.log_files:
+            raise UnsupportedPluginError("No Splashtop log files found on target")
+
+    @export(record=RemoteAccessLogRecord)
+    def logs(self) -> Iterator[RemoteAccessLogRecord]:
+        """Parse Splashtop log files.
+
+        Splashtop is a remote desktop application that can be used to get (persistent) access to a machine.
+        It might be used in combination with Atera Management Agent.
+
+        Refrences:
+            - https://www.synacktiv.com/en/publications/legitimate-rats-a-comprehensive-forensic-analysis-of-the-usual-suspects#atera-and-splashtop
+        """
+        target_tz = self.target.datetime.tzinfo
+
+        for log_file in self.log_files:
+            for ts, line in year_rollover_helper(log_file, RE_TS, "%b%d %H:%M:%S.%f", target_tz):
```

Comment:
This `year_rollover_helper` does a lot of things, like opening and reading the file :shrug: 
Should we handle errors when that fails or just crash and burn?